src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc: GroupsMap<T>, item: T) => {
+    const groupKey = item[key] as unknown as string;
+
+    if (!acc[groupKey]) {
+      acc[groupKey] = [];
+    }
+
+    acc[groupKey].push(item);
+
+    return acc;
+  }, {} as GroupsMap<T>);
```

Comment:
The initial value for reduce is being cast to `GroupsMap<T>`, which is correct, but you could also provide a more specific initial value to avoid casting. For example, you could use `Object.create(null)` to ensure the accumulator is a pure object without inherited properties, which can be safer and more predictable when using it as a map.